sensor/common/detector/metrics/metrics.go

```diff
@@ -120,8 +120,61 @@ var (
 		Name:      "detector_network_flows_dropped_total",
 		Help:      "A counter of the total number of network flows that were dropped if the detector buffer was full",
 	})
+
+	detectorBlockScanCalls = prometheus.NewCounterVec(prometheus.CounterOpts{
+		Namespace: metrics.PrometheusNamespace,
+		Subsystem: metrics.SensorSubsystem.String(),
+		Name:      "block_scan_calls",
+		Help:      "A counter that tracks the operations in blocking scan calls",
+	}, []string{"Operation", "Path"})
+
+	scanCallDuration = prometheus.NewHistogramVec(prometheus.HistogramOpts{
+		Namespace: metrics.PrometheusNamespace,
+		Subsystem: metrics.SensorSubsystem.String(),
+		Name:      "scan_call_duration",
+		Help:      "Time taken to call scan in milliseconds",
+		Buckets:   prometheus.ExponentialBuckets(4, 2, 8),
+	}, []string{})
+
+	scanAndSetCall = prometheus.NewCounterVec(prometheus.CounterOpts{
+		Namespace: metrics.PrometheusNamespace,
+		Subsystem: metrics.SensorSubsystem.String(),
+		Name:      "scan_and_set_call",
+		Help:      "A counter that tracks the operations in scan and set",
+	}, []string{"Operation"})
 )
 
+func AddBlockingScanCall(path string) {
+	detectorBlockScanCalls.With(prometheus.Labels{
+		"Operation": metrics.Add.String(),
+		"Path":      path,
+	}).Inc()
+}
+func RemoveBlockingScanCall() {
+	detectorBlockScanCalls.With(prometheus.Labels{
+		"Operation": metrics.Remove.String(),
+		"Path":      "",
+	}).Inc()
+}
+
+func SetScanCallDuration(duration time.Time) {
+	now := time.Now().UnixNano()
+	diff := now - duration.UnixNano()
+	scanCallDuration.With(prometheus.Labels{}).Observe(float64(diff / int64(time.Millisecond)))
+}
```

Comment:
I got a bit confused here. How about letting the time pkg do that conversion?

```suggestion
func SetScanCallDuration(start time.Time) {
	now := time.Now()
	durMilli := now.Sub(start).Milliseconds()
	scanCallDuration.With(prometheus.Labels{}).Observe(float64(durMilli))
}
```

Reply:
Done